pkg/controller/supportbundlecollection/controller.go

```diff
@@ -0,0 +1,986 @@
+	externalNodeListerSynced            cache.InformerSynced
+
+	// queue maintains the ExternalNode objects that need to be synced.
+	queue workqueue.RateLimitingInterface
+
+	// supportBundleCollectionStore is the storage where the populated internal support bundle collections are stored.
+	supportBundleCollectionStore storage.Interface
+	// supportBundleCollectionAppliedToStore is the storage where the required Nodes or ExternalNodes of a
+	// SupportBundleCollection are stored.
+	supportBundleCollectionAppliedToStore cache.Indexer
+	// supportBundleCollectionMutex protects the supportBundleCollectionStore from concurrent access
+	// during updates to the internal supportBundleCollection object.
+	supportBundleCollectionMutex sync.RWMutex
+}
+
+func NewSupportBundleCollectionController(
+	kubeClient kubernetes.Interface,
+	crdClient clientset.Interface,
+	supportBundleInformer crdinformers.SupportBundleCollectionInformer,
+	nodeInformer coreinformers.NodeInformer,
+	externalNodeInformer crdinformers.ExternalNodeInformer,
+	supportBundleCollectionStore storage.Interface) *Controller {
+	c := &Controller{
+		kubeClient: kubeClient,
+		crdClient:  crdClient,
+
+		supportBundleCollectionInformer:     supportBundleInformer,
+		supportBundleCollectionLister:       supportBundleInformer.Lister(),
+		supportBundleCollectionListerSynced: supportBundleInformer.Informer().HasSynced,
+		nodeLister:                          nodeInformer.Lister(),
+		nodeListerSynced:                    nodeInformer.Informer().HasSynced,
+		externalNodeLister:                  externalNodeInformer.Lister(),
+		externalNodeListerSynced:            externalNodeInformer.Informer().HasSynced,
+		queue:                               workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "supportBundleCollection"),
+		supportBundleCollectionStore:        supportBundleCollectionStore,
+		supportBundleCollectionAppliedToStore: cache.NewIndexer(getSupportBundleCollectionKey, cache.Indexers{
+			nodeSelectorIndex:                nodeSelectorIndexFunc,
+			allNodesIndex:                    allNodesIndexFunc,
+			nodeNameIndex:                    nodeNameIndexFunc,
+			externalNodeSelectorIndex:        externalNodeSelectorIndexFunc,
+			allExternalNodesInNamespaceIndex: allExternalNodesInNamespaceIndexFunc,
+			externalNodeNameIndex:            externalNodeNameIndexFunc,
+			processingNodesIndex:             processingNodesIndexFunc,
+			processingExternalNodesIndex:     processingExternalNodesIndexFunc,
+		}),
+	}
+	c.supportBundleCollectionInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addSupportBundleCollection,
+			UpdateFunc: c.updateSupportBundleCollection,
+			DeleteFunc: c.deleteSupportBundleCollection,
+		},
+		resyncPeriod)
+	nodeInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addNode,
```

Comment:
Not just Node update. I was asking could we remove all Node events inc. add/delete, assuming once collection started, we ignore any Node changes.

Reply:
Since we support using label to select Node/ExternalNode, if we ignore the changes on Node/ExternalNode, I think it is possible to introduce mismatch issues.

Reply:
@tnqn @reachjainrahul Would you share your thoughts?

Reply:
Again, anyway you cannot cover all cases. We can declare we just include Nodes exist when the collection starts.

Reply:
Got it, then I would remove this.